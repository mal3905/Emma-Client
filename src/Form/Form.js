import React from 'react'
import './Form.css'


export default function Form(props) {
  const { className, ...otherProps } = props
  return (
    <form
      className={['Emma-form', className].join(' ')}
      action='#'
      {...otherProps}
    />
  )
}