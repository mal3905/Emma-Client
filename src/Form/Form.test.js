import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Form from '../Form/Form'

describe(`Form component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.Form by default', () => {
    const wrapper = shallow(<Form />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Form given props', () => {
    const wrapper = shallow(<Form {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
