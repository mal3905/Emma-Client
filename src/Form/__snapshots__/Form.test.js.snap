// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders a form.Form by default 1`] = `
<form
  action="#"
  className="Emma-form "
/>
`;

exports[`Form component renders the Form given props 1`] = `
<form
  action="#"
  className="Emma-form test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</form>
`;
